lib/SimpleSAML/Locale/Translate.php

```diff
@@ -108,7 +108,7 @@ private function getDictionary($name)
                 $fileName = substr($name, $sepPos + 1);
                 $dictDir = Module::getModuleDir($module).'/dictionaries/';
             } else {
-                $dictDir = $this->configuration->getPathValue('dictionarydir', 'dictionaries/');
+                $dictDir = $this->configuration->getPathValue('dictionarydir', 'dictionaries/') ?: 'dictionaries/';
```

Comment:
This may look weird, but if the configuration-setting is set to `null`, then Configuration::getPathValue will return `null` rather than the default value...

Reply:
Agreed that it's a bug in `getPathValue`, but I think the solution is not good either. I'd rather throw an exception if `null === $dictDir`. Your configuration should not work when you set things to `null` that cannot be null.

Reply:
I don't think I agree on a setting that is so fundamental..
`null` can also mean 'unset' in this case and if such a fundamental setting is unset,, a sane default should fill in.. It's just the that the second param of `getPathValue` isn't the safe default in every case..
In any case, this may not be the best fix, but it is what it is until 2.0 at least..
I guess in a future release a bool-param that specifies whether or not a value is mandatory could fix this..

Reply:
The `dictionarydir` setting is indeed fundamental, that's why the software should crash if it's overridden to an invalid value. `null` is null, 'unset' is unset and a string is a string. When it's a string, use the string. When it's unset, use the default. When it's `null`, crash with a sensible error message.